src/DependencyGraph/DependencyGraph.ts

```diff
@@ -125,15 +128,15 @@ export class DependencyGraph {
     }
 
     this.correctInfiniteRangesDependency(address)
+
+    return this.getAndClearContentChanges()
   }
 
-  public setCellEmpty(address: SimpleCellAddress) {
-    const vertex = this.addressMapping.getCell(address)
+  public setCellEmpty(address: SimpleCellAddress): ContentChanges {
+    const vertex = this.shrinkPossibleMatrixAndGetCell(address)
     if (vertex === null) {
-      return
+      return ContentChanges.empty()
```

Comment:
especially that we assume that there are no changes, but could matrix shrinking generate content changes?